interfaces/mount/sorting.go

```diff
@@ -0,0 +1,42 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package mount
+
+import (
+	"strings"
+)
+
+// byMagicDir allows sorting an array of entries that automagically assumes
+// each entry ends with a trailing slash.
```

Comment:
This idea came from @pedronis. The path without the slash will obviously sort but then will not have the desired semantics. If we resort to regular sort the following failure is reported:

```
FAIL: change_test.go:139: changeSuite.TestNeededChangesSmartEntryComparison
 [...]
... obtained []mount.Change = []mount.Change{mount.Change{Entry:mount.Entry{Name:"", Dir:"/a/b-1/3", Type:"", Options:[]string(nil), DumpFrequency:0, CheckPassNumber:0}, Action:"umount"}, mount.Change{Entry:mount.Entry{Name:"/dev/sda1", Dir:"/a/b", Type:"", Options:[]string(nil), DumpFrequency:0, CheckPassNumber:0}, Action:"umount"}, mount.Change{Entry:mount.Entry{Name:"/dev/sda2", Dir:"/a/b", Type:"", Options:[]string(nil), DumpFrequency:0, CheckPassNumber:0}, Action:"mount"}}
... expected []mount.Change = []mount.Change{mount.Change{Entry:mount.Entry{Name:"", Dir:"/a/b/c", Type:"", Options:[]string(nil), DumpFrequency:0, CheckPassNumber:0}, Action:"umount"}, mount.Change{Entry:mount.Entry{Name:"/dev/sda1", Dir:"/a/b", Type:"", Options:[]string(nil), DumpFrequency:0, CheckPassNumber:0}, Action:"umount"}, mount.Change{Entry:mount.Entry{Name:"", Dir:"/a/b-1/3", Type:"", Options:[]string(nil), DumpFrequency:0, CheckPassNumber:0}, Action:"umount"}, mount.Change{Entry:mount.Entry{Name:"/dev/sda2", Dir:"/a/b", Type:"", Options:[]string(nil), DumpFrequency:0, CheckPassNumber:0}, Action:"mount"}, mount.Change{Entry:mount.Entry{Name:"", Dir:"/a/b/c", Type:"", Options:[]string(nil), DumpFrequency:0, CheckPassNumber:0}, Action:"mount"}}
```

If you analyze what happens there instead of:

```
    c.Assert(changes, DeepEquals, []mount.Change{
        {Entry: mount.Entry{Dir: "/a/b/c"}, Action: mount.Unmount},
        {Entry: mount.Entry{Dir: "/a/b", Name: "/dev/sda1"}, Action: mount.Unmount},
        {Entry: mount.Entry{Dir: "/a/b-1/3"}, Action: mount.Unmount},

        {Entry: mount.Entry{Dir: "/a/b", Name: "/dev/sda2"}, Action: mount.Mount},
        {Entry: mount.Entry{Dir: "/a/b/c"}, Action: mount.Mount},
    })
````

We get

```
[]mount.Change{
   {Entry:mount.Entry{Dir:"/a/b-1/3"}, Action:"umount"},
   {Entry:mount.Entry{Dir:"/a/b", Name:"/dev/sda1"}, Action:"umount"},
   {Entry:mount.Entry{Dir:"/a/b", Name:"/dev/sda2"}, Action:"mount"}}
```

We expect the following sort order on those directories:
- `/a/b-1`
- `/a/b-1/3`
- `/a/b`
- `/a/b/c`

But because `-` sorts before `/` this is not what we are getting. The solution is to insert the trailing slash so that we get the desired order.